common/searchattribute/encode_value.go

```diff
@@ -45,15 +45,52 @@ func EncodeValue(val interface{}, t enumspb.IndexedValueType) (*commonpb.Payload
 	return valPayload, nil
 }
 
+func DecodeValue(value *commonpb.Payload, t enumspb.IndexedValueType) (interface{}, error) {
+	return decodeValueInternal(value, t, true)
+}
+
+func DecodeValueStrict(value *commonpb.Payload, t enumspb.IndexedValueType) (any, error) {
```

Comment:
I would avoid `Strict` word here. It is too generic. I like `allowList` more, as it exactly describes what it is strict about. On a caller side you have the opposite code, which converts bool var into the one or another method call. Why not just to add `allowList bool` to `DecodeValue` and just pass it down from the caller?